### PR TITLE
Added inmemory store.

### PR DESCRIPTION
**Pull request by ouphi**
Added inmemory store

### CODE REVIEW

File: app/backend/store/inmemory_image_store.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    """
+        InMemory store to store images.
+    """
+    images = []
+
+    def store_image(self, image_content: bytes, image_description: str):
+        """ Store an image and description."""
+        self.images.append((image_content, image_description))
+
+    def get_random_image(self):
```
Comment (ouphi) [inline app/backend/store/inmemory_image_store.py:18]: => add get_image_description(self, image_id)

for api logic:
=> verify_guess(image_id, description)

File: app/backend/store/inmemory_image_store.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        self.images = {}
+
+    def store_image(self, image_content: bytes, image_description: str):
+        """ Store an image and description."""
+        image_id = uuid.uuid1()
+        self.images[image_id] = (image_content, image_description)
+
+    def get_random_image(self):
+        """ Get a random image."""
+        if len(self.images.keys()) == 0:
```
Comment (gukoff) [inline app/backend/store/inmemory_image_store.py]:
```suggestion
        if not images: 
```

File: app/backend/store/inmemory_image_store.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        """ Store an image and description."""
+        image_id = uuid.uuid1()
+        self.images[image_id] = (image_content, image_description)
+
+    def get_random_image(self):
+        """ Get a random image."""
+        if len(self.images.keys()) == 0: 
+            return None
+        image_id = random.choice(list(self.images.keys()))
+        return (image_id, self.images[image_id][0])
```
Comment (gukoff) [inline app/backend/store/inmemory_image_store.py]:
```suggestion
        image_content, image_description = self.images[image_id]
        return (image_id, image_content)
```

File: app/backend/store/inmemory_image_store.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    def get_image_description(self, image_id):
+        image = self.images.get(image_id)
+        return None if image is None else image[1]
```
Comment (gukoff) [inline app/backend/store/inmemory_image_store.py]:
```suggestion
        if image_id not in self.images:
            return None
        image_content, image_description = self.images[image_id]
        return image_description
```
I think that if we're more verbose, the code will be more readable to the students. What do you think?

File: app/backend/tests/test_inmemory_image_store.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        store = InMemoryImageStore()
+        store.images = {"id1": ("image_content_1", "cat")}
+        image_description = store.get_image_description("id1")
+        assert image_description == "cat"
+
+    @staticmethod
+    def test_get_image_description_empty_store():
+        store = InMemoryImageStore()
+        image_description = store.get_image_description("id1")
+        assert image_description is None
```
Comment (gukoff) [inline app/backend/tests/test_inmemory_image_store.py]:
```suggestion
def test_get_random_image_when_empty_store():
    store = InMemoryImageStore()
    assert store.get_random_image() == None

def test_get_random_image_when_one_image():
    store = InMemoryImageStore()
    image = ("base64image", "a cat")
    store.store_image(image[0], image[1])
    assert store.get_random_image()[1] == image[0]

def test_get_image_description():
    store = InMemoryImageStore()
    store.images = {"id1": ("image_content_1", "cat")}
    image_description = store.get_image_description("id1")
    assert image_description == "cat"

def test_get_image_description_empty_store():
    store = InMemoryImageStore()
    image_description = store.get_image_description("id1")
    assert image_description is None

```

I didn't check, but I'm pretty sure you can remove the class altogether

**Pull request merged**